Use consistently `count` and `indexes` for band listing and counting (removing `nbands` and  `bands`)

rhugonnet opened the pull request:
To be fully consistent with `rasterio`.

Resolves #280

adehecq commented on geoutils/georaster/raster.py:
```diff
@@ -2331,19 +2318,19 @@ def interp_points(
         # rpts = np.array(rpts)
 
-    def split_bands(self: RasterType, copy: bool = False, subset: list[int] | int | None = None) -> list[Raster]:
+    def split_indexes(self: RasterType, copy: bool = False, subset: list[int] | int | None = None) -> list[Raster]:
```
I think in this case, it makes more sense to keep "band" as it is much clearer what this will do. To me, indexes refer rather to the position of the bands, so it does not make sense to split them.

rhugonnet replied:
Agree, changed

adehecq reviewed (commented):
Hopefully it will be much clearer now! 
But what I don't get is that basically here you just removed nbands into count and bands into indexes. But I thought we had a different use of count and nbands, depending on what was loaded and what was on disk, no? 
In the issue, I suggested using a `count_loaded` and `indexes_loaded` but it seems like it was not needed, why?

rhugonnet commented:
My bad, I forgot that point! I added them back and added tests for them.

The names `count_loaded` and `indexes_loaded` sounded good. But then `count` and `index` are "on-disk", and can take a value of `None`, which is not so practical when manipulating in-memory-only `Raster`. So instead I went for `count_on_disk` and `indexes_on_disk`, while `count` and `indexes` are in-memory. Clarified everything in the descriptions.

adehecq commented:
> My bad, I forgot that point! I added them back and added tests for them.
> 
> The names `count_loaded` and `indexes_loaded` sounded good. But then `count` and `index` are "on-disk", and can take a value of `None`, which is not so practical when manipulating in-memory-only `Raster`. So instead I went for `count_on_disk` and `indexes_on_disk`, while `count` and `indexes` are in-memory. Clarified everything in the descriptions.

Sounds good. Everything seems perfect now, thanks !
